### PR TITLE
Reduce cheribsdtest verbosity

### PR DESCRIPTION
**Pull request by bsdjhb**
- cheribsdtest: Remove the sig_dfl_neq_ign test
- cheribsdtest: Trim some spurious debug printfs.
- cheribsdtest: Export verbose flag so tests can use it
- cheribsdtest: Move debug printfs conditional on verbose
- cheribsdtest: Remove some stale macros for printing capabilities
- run-many: Hide SKIP test results

### CODE REVIEW

File: bin/cheribsdtest/cheribsdtest.c [hunk 1]
```diff
@@ -91,7 +91,7 @@ static int fast_tests_only;
 static int qtrace;
 static int qtrace_user_mode_only;
 static int sleep_after_test;
-static int verbose;
+int verbose;
```
Comment (jrtc27) [inline bin/cheribsdtest/cheribsdtest.c:94]: Is this worth moving to a separate block?

Comment (bsdjhb) [reply bin/cheribsdtest/cheribsdtest.c:94]: I can move it.  Either way is fine with me.

File: bin/cheribsdtest/cheribsdtest_syscall.c [hunk 2]
```diff
@@ -62,30 +62,6 @@
...
-CHERIBSDTEST(sig_dfl_neq_ign, "Test SIG_DFL != SIG_IGN")
-{
-	void * __capability sic = (__cheri_tocap void * __capability)SIG_IGN;
-
-	/*
-	 * This may appear redundant, but it tickles some of the optimizers
-	 * differently than the condition below, which, apparently, can get
-	 * constant-folded and DCE'd, while this does not.
-	 */
-	int eq = (SIG_IGN == SIG_DFL);
```
Comment (arichardson) [inline bin/cheribsdtest/cheribsdtest_syscall.c:74]: This is checking that SIG_IGN is defined in a way that doesn't use CToPtr/CFromPtr, so it is more of an OS headers test than a compiler test. git blame points at https://github.com/CTSRD-CHERI/cheribsd/commit/1e704e0f3837a5de493598449408de62ff6a8db6

Comment (jrtc27) [reply bin/cheribsdtest/cheribsdtest_syscall.c:74]: The talk of constant-folding is a bit compiler-facing, though, and fragile. Probably if we really want to make sure we are careful we should just be using volatile variables to capture the actual value of each constant.

Comment (arichardson) [reply bin/cheribsdtest/cheribsdtest_syscall.c:74]: I agree this test is rather useless now that the underlying issue has been fixed. However, I do wonder if we would find the same issue if we were to port cheribsdtest to e.g. linux.

Comment (jrtc27) [reply bin/cheribsdtest/cheribsdtest_syscall.c:74]: https://git.morello-project.org/morello/kernel/linux/-/blob/morello/master/include/uapi/asm-generic/signal-defs.h#L88-90 🙃

Comment (bsdjhb) [reply bin/cheribsdtest/cheribsdtest_syscall.c:74]: Could we safely keep a version that is just the tail end of the test that requires the constants to test !=?

Comment (jrtc27) [reply bin/cheribsdtest/cheribsdtest_syscall.c:74]: That should be fine, it's able to detect the buggy case from before https://cheri-compiler-explorer.cl.cam.ac.uk/z/EMnYEW

**Review by jrtc27 — APPROVED** [review]

**Review by arichardson — APPROVED** [review]

**Pull request merged**